include/internal/ktls.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef HEADER_INTERNAL_KTLS
+#define HEADER_INTERNAL_KTLS
+
+#if defined(OPENSSL_SYS_LINUX)
+# include <linux/version.h>
+
+#define K_MAJ   4
+#define K_MIN1  13
+#define K_MIN2  0
+#if LINUX_VERSION_CODE < KERNEL_VERSION(K_MAJ, K_MIN1, K_MIN2)
+
+# ifndef PEDANTIC
+#  warning "KTLS requires Kernel Headers >= 4.13.0"
+#  warning "Skipping Compilation of KTLS data path"
+# endif
+
+#define TLS_TX                  1       /* Set transmit parameters */
+
+#define TLS_CIPHER_AES_GCM_128                          51
+#define TLS_CIPHER_AES_GCM_128_IV_SIZE                  8
+#define TLS_CIPHER_AES_GCM_128_KEY_SIZE         16
+#define TLS_CIPHER_AES_GCM_128_SALT_SIZE                4
+#define TLS_CIPHER_AES_GCM_128_TAG_SIZE         16
+#define TLS_CIPHER_AES_GCM_128_REC_SEQ_SIZE             8
+
+#define TLS_SET_RECORD_TYPE     1
+
+struct tls_crypto_info {
+    unsigned short version;
+    unsigned short cipher_type;
+};
+
+struct tls12_crypto_info_aes_gcm_128 {
+    struct tls_crypto_info info;
+    unsigned char iv[TLS_CIPHER_AES_GCM_128_IV_SIZE];
+    unsigned char key[TLS_CIPHER_AES_GCM_128_KEY_SIZE];
+    unsigned char salt[TLS_CIPHER_AES_GCM_128_SALT_SIZE];
+    unsigned char rec_seq[TLS_CIPHER_AES_GCM_128_REC_SEQ_SIZE];
+};
+
+
+/* Dummy functions here */
+ossl_inline int ktls_enable(int fd)
```

Comment:
Travis failures are due to this I think:
````
ssl/t1_enc.o: In function `ktls_enable':
t1_enc.c:(.text+0x280): multiple definition of `ktls_enable'
ssl/ssl_lib.o:ssl_lib.c:(.text+0x4c0): first defined here
collect2: error: ld returned 1 exit status
make[1]: *** [libssl.so] Error 1
make: *** [all] Error 2
+/// MAKE FAILED
````
Probably this should be declared `static` otherwise it gets defined multiple times in every file that includes this header.

Reply:
Done. Thanks.